paddle/framework/attribute.h

```diff
@@ -29,11 +29,13 @@ namespace framework {
 
 typedef boost::variant<boost::blank, int, float, std::string, std::vector<int>,
                        std::vector<float>, std::vector<std::string>,
-                       std::vector<std::pair<int, int>>>
+                       std::vector<std::pair<int, int>>, BlockDesc>
     Attribute;
 
 typedef std::unordered_map<std::string, Attribute> AttributeMap;
 
+static ProgramDesc g_program_desc;
```

Comment:
In Google C++ style, a class instance cannot be a global variable.

Reply:
It cannot be written in .h

Reply:
Variable definition usually should not be written in a header file, because it will define all variables in .cc which includes this header. Here should be a declaration.

* To define an int value, we use `int a;`. To declare an int value, we use `extern int a;`. `extern` means somewhere is define that int value, but every `.cc` which includes this header can use that int value.

* `static` means not to generate symbol name in `.S` or `.o` file. So this will actually create many `ProgramDesc`, each `.cc` has a new `g_program_desc`.

Reply:
Got it